ui/src/main/kotlin/edu/stanford/spezi/ui/ImageResource.kt

```diff
@@ -30,4 +39,50 @@ sealed interface ImageResource {
     ) : ImageResource {
         override val identifier = UUID().toString()
     }
+
+    @Composable
+    override fun Content(modifier: Modifier) {
+        Content(modifier = modifier, tint = Colors.primary)
+    }
+
+    @Composable
+    fun Content(modifier: Modifier, tint: Color) {
```

Comment:
Consider introducing this typealias https://github.com/StanfordSpezi/SpeziKt/pull/143/files#diff-5e1b1e9082809c2ba95dc5c5fbdf0b7d44d4167b9c9e8ea87dd50e1076a21290R19

See doc comments there, you can have than `tint: ComposeValue<Color> get() { Colors.primary } as an interface requirement with default value, this way we can stick to the base ComposableContent interface, and can extend the type instead of introducing new methods for every config option

Reply:
I imagine ImageResource to not really make assumptions about the appearance in UI, but more like an object to know how to get some image data... In the UserProfile for example, we want to use the image as a mask and then apply a custom tint that isn't provided by an app developer but instead set by spezi:ui-personalinfo.

We may want to think about having two interfaces `ComposableContent` and `ConfigurableComposableContent<Configuration>` maybe, where one allows to have a generic constraint with a given configuration to set some values during composition. In this case, we could then create a data class `ImageResourceConfiguration` with the tint as one of the properties - or sth like that.